src/actions/noting/remove-note.js

```diff
@@ -0,0 +1,38 @@
+var isVFocus = require('../../utils/vfocus/is-vfocus');
+var errorHandle = require('../../utils/error-handle');
+
+var findScribeMarkers = require('../../utils/noting/find-scribe-markers');
+var noteCache = require('../../utils/noting/note-cache');
+var findParentNoteSegment = require('../../utils/noting/find-parent-note-segment');
+var findNoteById = require('../../utils/noting/find-note-by-id');
+var unWrapNote = require('./unwrap-note');
+var ensureNoteIntegrity = require('./ensure-note-integrity');
+
+// treeFocus: tree focus of tree containing two scribe markers
+// Note that we will mutate the tree.
+module.exports = function removeNote(focus){
+
+  if (!isVFocus(focus)){
+    errorHandle('Only a valid VFocus element can be passed to removeNote, you passed: %s', focus);
+  }
+
+  // We assume the caller knows there's only one marker.
+  var marker = findScribeMarkers(focus)[0];
+
+
+  // We can't use findEntireNote here since it'll sometimes give us the wrong result.
+  // See `findEntireNote` documentation. Instead we look the note up by its ID.
+  noteCache.set(focus);
```

Comment:
My understanding is that ```noteCache.set``` requires the focus to rebuild
the tree and cache it. So it's not technically caching the focus right?
Maybe the function should be renamed.

On 14 January 2015 at 11:29, Jon Parsons <notifications@github.com> wrote:

> In src/actions/noting/remove-note.js
> <https://github.com/guardian/scribe-plugin-noting/pull/65#discussion-diff-22930976>
> :
>
> > +
> > +// treeFocus: tree focus of tree containing two scribe markers
> > +// Note that we will mutate the tree.
> > +module.exports = function removeNote(focus){
> > +
> > +  if (!isVFocus(focus)){
> > +    errorHandle('Only a valid VFocus element can be passed to removeNote, you passed: %s', focus);
> > +  }
> > +
> > +  // We assume the caller knows there's only one marker.
> > +  var marker = findScribeMarkers(focus)[0];
> > +
> > +
> > +  // We can't use findEntireNote here since it'll sometimes give us the wrong result.
> > +  // See `findEntireNote` documentation. Instead we look the note up by its ID.
> > +  noteCache.set(focus);
>
> This was in the original code base.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/scribe-plugin-noting/pull/65/files#r22930976>
> .
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396